reconcile/openshift_tekton_resources.py

```diff
@@ -0,0 +1,353 @@
+                            'limits': {'cpu': '200m',
+                                       'memory': '300Mi'}}
+SAAS_FILES_QUERY = """
+{
+  saas_files: saas_files_v2 {
+    path
+    name
+    configurableResources
+    pipelinesProvider {
+      name
+      provider
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
+      }
+    }
+  }
+}
+"""
+
+
+class OpenshiftTektonResourcesNameTooLongError(Exception):
+    pass
+
+
+class OpenshiftTektonResourcesBadConfigError(Exception):
+    pass
+
+
+# Builds a list of v2 saas files from qontract-server
+def get_saas_files(saas_file_name: Optional[str]) -> list[dict[str, Any]]:
+    saas_files = [
+        s for s in gql.get_api().query(SAAS_FILES_QUERY)['saas_files']
+        if s.get('configurableResources')]
+
+    if saas_file_name:
+        saas_file = None
+        for s in saas_files:
+            if s['name'] == saas_file_name:
+                saas_file = s
+                break
+
+        return [saas_file] if saas_file else []
+
+    return saas_files
+
+
+def fetch_tkn_providers(saas_files: Iterable[dict[str, Any]]) \
+        -> dict[str, Any]:
+    duplicates: set[str] = set()
+    all_tkn_providers = {}
+    for pipeline_provider in queries.get_pipelines_providers():
+        if pipeline_provider['provider'] != 'tekton':
+            continue
+
+        if pipeline_provider['name'] in all_tkn_providers:
+            duplicates.add(pipeline_provider['name'])
+        else:
+            all_tkn_providers[pipeline_provider['name']] = pipeline_provider
+
+    if duplicates:
+        raise OpenshiftTektonResourcesBadConfigError(
+            'There are duplicates in tekton providers names: '
+            f'{", ".join(duplicates)}')
+
+    # Only get the providers that are used by the saas files
+    # Add the saas files belonging to it
+    tkn_providers = {}
+    for saas_file in saas_files:
+        tkn_provider = saas_file['pipelinesProvider']['name']
+        if tkn_provider not in tkn_providers:
+            tkn_providers[tkn_provider] = all_tkn_providers[tkn_provider]
+
+        if 'saas_files' not in tkn_providers[tkn_provider]:
+            tkn_providers[tkn_provider]['saas_files'] = []
+
+        tkn_providers[tkn_provider]['saas_files'].append(saas_file)
+
+    return tkn_providers
+
+
+# Create an array of dicts that will be used as args of ri.add_desired
+# This will also add resourceNames inside tkn_providers['namespace']
+# while we are migrating from the current system to this integration
+def fetch_desired_resources(tkn_providers: dict[str, Any]) \
+        -> list[dict[str, Union[str, OR]]]:
+    desired_resources = []
+    for tkn_provider in tkn_providers.values():
+        namespace = tkn_provider['namespace']['name']
+        cluster = tkn_provider['namespace']['cluster']['name']
+        deploy_resources = tkn_provider.get('deployResources',
```

Comment:
Does this `tkn_provider` come from the graphql query you mention above? If so, please consider `tkn_provider.get('deployResources') or DEFAULT_DEPLOY_RESOURCES`, see [this ticket](https://issues.redhat.com/browse/APPSRE-3868).

Reply:
I will change it.